behandlingsprosess/src/main/java/no/nav/ung/sak/domene/behandling/steg/registerinntektkontroll/KontrollerInntektSteg.java

```diff
@@ -86,18 +78,26 @@ public KontrollerInntektSteg() {
 
         var rapporterteInntekterTidslinje = rapportertInntektMapper.mapAlleGjeldendeRegisterOgBrukersInntekter(behandlingId);
         var prosessTriggerTidslinje = prosessTriggerPeriodeUtleder.utledTidslinje(behandlingId);
-        var uttalelser = registerinntektUttalelseTjeneste.hentUttalelser(behandlingId);
-        var registerinntekterForIkkeGodkjentUttalelse = rapportertInntektMapper.finnRegisterinntekterForUttalelse(behandlingId, uttalelser);
-        var kontrollResultat = KontrollerInntektTjeneste.utførKontroll(prosessTriggerTidslinje, rapporterteInntekterTidslinje, registerinntekterForIkkeGodkjentUttalelse);
-        håndterPeriodisertKontrollresultat(kontekst, kontrollResultat, rapporterteInntekterTidslinje, prosessTriggerTidslinje);
+        var etterlysninger = etterlysningRepository.hentEtterlysninger(kontekst.getBehandlingId(), EtterlysningType.UTTALELSE_KONTROLL_INNTEKT);
+
+        var etterlysningsperioder = etterlysninger.stream()
+            .map(it -> new EtterlysningsPeriode(
+                it.getPeriode().toLocalDateInterval(),
+                new EtterlysningInfo(it.getStatus(), it.getUttalelse() != null ? it.getUttalelse().erEndringenGodkjent() : null),
```

Comment:
Eg ser ikkje grunnen til å wrappe dette. Kvifor ikkje berre ha det på perioden?

Reply:
Eg ser no at du kopierer med denne informasjonen videre, men eg synes det var litt merkelig å wrappe akkurat denne infoen her. Perioden er vel også relevant?

Reply:
fordi den senere brukes til å koble sammen EtterlysningOgRegisterInntekt uten periode og uten IAYGrunnalgRef. 

... jeg fulgte egentlig måten det var gjort for BrukersUttalelsePeriode og BrukersUttalelseForRegister - men jeg tenkte litt på det samme. Koden for å finne inntekt fra referansen var sterkt knyttet til de to klassene.